client/my-sites/marketplace/pages/marketplace-product-install/index.tsx

```diff
@@ -170,14 +170,22 @@ const MarketplaceProductInstall = ( {
 		}
 	} );
 
+	const { primaryDomain } = useSelector( getPurchaseFlowState );
+
+	const shouldShowNoDirectAccessError =
+		// 1. This is a plugin upload flow (via zip file) and we don't have a primary domain set
+		( isPluginUploadFlow && ! primaryDomain ) ||
```

Comment:
Do we really care about the `primaryDomain` or can we simplify the logic and avoid showing the error for any `isPluginUploadFlow`?

Reply:
Good point, I added this extra condition in a later commit to show this screen when the user navigates back or accesses directly to the URL `/marketplace/plugin/install/:siteDomain` because they will get stuck in this screen otherwise: 

![CleanShot 2025-02-07 at 13 34 09@2x](https://github.com/user-attachments/assets/29527077-9296-44ee-92ef-4ed8c8ccf264)